### ISSUE TITLE
Disabled button still submits form.

### ISSUE THREAD

**Issue by hbowden** [DigitalRiver/react-atlas#689]
When a submit button is disabled you should not be able to submit the form. However the form still submits. Below is some example code of the bug.

```
<Form action={"/some/random/url"}>
  <TextField small placeholder="Text" type="text" name="textName"></TextField>
  <Button disabled={true} type="submit">Submit</Button>
</Form>
```

**Comment by Raymans** [comment]
#696 Don't trigger onClick and type="submit" if it's disabled button.

This should fix disabled Button submit issue.

**Comment by stuller** [comment]
fixed with PR #696
